server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java

```diff
@@ -383,16 +382,12 @@ public void testChildrenTasksCancelledOnTimeout() throws Exception {
         ActionFuture<TestResponse> rootTaskFuture = client().execute(TransportTestAction.ACTION, rootRequest);
         allowEntireRequest(rootRequest);
         waitForRootTask(rootTaskFuture, true);
-        assertBusy(() -> {
-            for (DiscoveryNode node : nodes) {
-                TransportService transportService = internalCluster().getInstance(TransportService.class, node.getName());
-                for (ThreadPoolStats.Stats stat : transportService.getThreadPool().stats()) {
-                    assertEquals(0, stat.getActive());
-                    assertEquals(0, stat.getQueue());
-                }
-            }
-        }, 60L, TimeUnit.SECONDS);
         ensureBansAndCancellationsConsistency();
+
+        // Make sure all descendent requests have completed
+        for (TestRequest subRequest : rootRequest.descendants()) {
+            assertTrue(completedLatches.get(subRequest).await(60, TimeUnit.SECONDS));
```

Comment:
60s seems like a long wait per latch in a test. Could we use `ESTestCase#safeAwait(CountDownLatch)` which has a 10s timeout? (Also means there's no checked exceptions here too)

Reply:
@DaveCTurner I did not know about this one, thanks! Done.